tests/utils.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+import os
+import subprocess
+from typing import TypeVar
+
+Self = TypeVar("Self", bound="Samarium")
+
+
+class Samarium:
+    def __init__(self, *, file: str) -> None:
+        self.file = file
+        self.proc: subprocess.Popen[bytes] | None = None
+        self._stdout: bytes | None = None
+        self._stderr: bytes | None = None
+
+    def __enter__(self: Self) -> Self:
+        path = self.file if os.name != 'nt' else self.file.replace("/", "\\")
```

Comment:
That's a bit awkward, imho. This would be technically more safe to use `os.path.join()`.

Reply:
I would but `abspath` (the more correct one here because paths are supplied as posix) has a massive skill issue so I'm not using pathlib